src/main/java/com/sonymobile/jenkins/plugins/mq/mqnotifier/MQConnection.java

```diff
@@ -159,26 +198,20 @@ private String getRoutingKey() {
      * @param body the message body
      */
     public void addMessageToQueue(String exchange, String routingKey, AMQP.BasicProperties props, byte[] body) {
-        synchronized (this) {
-            if (messageQueue == null) {
-                messageQueue = new LinkedBlockingQueue(MESSAGE_QUEUE_SIZE);
-            }
-            if (messageQueueThread == null || !messageQueueThread.isAlive()) {
-                messageQueueThread = new Thread(new Runnable() {
-                    @Override
-                    public void run() {
-                        sendMessages();
-                    }
-                });
-                messageQueueThread.start();
-                LOGGER.info("messageQueueThread recreated since it was null or not alive.");
-            }
+        if (messageQueueThread == null || !messageQueueThread.isAlive()) {
```

Comment:
Probably not a super big deal, but removing synchronized could lead to a situation where two threads a running simultaneously.

If you add two messages at the same time and the messageQueueThread is not alive (or null) both threads can and up on line #202. If one of the threads completes #202-#208 before the other thread has a chance to advanced you will have two threads running at the same time.

Reply:
Ah, yes, good point. Didn't think of the case when this method is called from separate threads.